integration-tests/src/mock_network/mod.rs

```diff
@@ -0,0 +1,287 @@
+use actix::{Actor, Context, Handler, Recipient};
+use near_chain::{Block, BlockHeader, Chain, ChainStoreAccess, Error};
+use near_chain_configs::GenesisConfig;
+use near_client::sync;
+use near_network::types::{
+    FullPeerInfo, NetworkClientMessages, NetworkInfo, NetworkRequests, NetworkResponses,
+    PeerManagerMessageRequest, PeerManagerMessageResponse,
+};
+use near_network_primitives::types::{
+    PartialEdgeInfo, PartialEncodedChunkRequestMsg, PartialEncodedChunkResponseMsg, PeerInfo,
+};
+use near_primitives::block::GenesisId;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{BlockHeight, ShardId};
+use std::collections::HashMap;
+use std::time::Duration;
+
+mod setup;
+
+/// MockPeerManagerActor mocks PeerManagerActor and responds to messages from ClientActor.
+/// Instead of sending these messages out to other peers, it simulates a network and reads
+/// the needed block and chunk content from storage.
+/// MockPeerManagerActor has the following responsibilities
+/// - Responds to BlockRequest, BlockHeadersRequest and PartialEncodedChunkRequest
+/// - Sends NetworkInfo to ClientActor periodically
+/// - Simulates block production and sends the most "recent" block to ClientActor
+struct MockPeerManagerActor {
+    /// Client address for the node that we are testing
```

Comment:
I'd also mention that this will be used to send the 'reponses' for the requests that we get.